util.js

```diff
@@ -0,0 +1,302 @@
+ * @param {Object} options - Additional options passed to the exec command.
+ * @param {string} ghAccessToken - GitHub access token for authentication.
+ * @throws Will throw an error if the execution fails.
+ * @example -> await gitClone('myRepo', 'main', './targetDir', {}, 'ghAccessToken');
+ */
+async function gitClone(repositoryName, branch, targetDir, options, ghAccessToken="" ) {
+    try {
+        const branchOption = branch ? ['-b', branch] : [];
+
+        const authPart = ghAccessToken ? `${ghAccessToken}@` : '';
+        const repoUrl = `https://${authPart}github.com/Pearson-Advance/${repositoryName}.git`;
+
+        await exec.exec(
+            'git',
+            [
+                'clone',
+                ...branchOption,
+                repoUrl,
+                targetDir,
+            ],
+            options
+        );
+    } catch (error) {
+        throw new Error(`Failed to clone repository: ${error.message}`);
+    }
+}
+  
+/**
+ * Handles the cloning and requirements setup for private repositories.
+ * @async
+ * @param {Array} privateRepositories - An array containing the names of the private repositories to be cloned.
+ * @param {Array} branches - An array containing the branches to be cloned from each repository.
+ * @param {string} ghAccessToken - GitHub access token to authenticate for private repository cloning.
+ * @param {string} tutorRoot - The root directory path where the cloned repositories will be placed.
+ * @param {Object} options - Execution options for shell commands.
+ */
+async function handlePrivatePackages(privateRepositories, branches, ghAccessToken, tutorRoot, options) {
+    core.info("Handling private packages...");
+    if (!Array.isArray(privateRepositories) || privateRepositories.length === 0) {
+        core.warning('No private repositories provided.');
+        return;
+    }
+    if (!Array.isArray(branches) || branches.length !== privateRepositories.length) {
+        core.error('Mismatch between number of repositories and number of branches.');
+        throw new Error('Branches and repositories array lengths must match.');
+    }
+    
+    const privateTxtPath = `${tutorRoot}/env/build/openedx/requirements/private.txt`;
+    const requirements = [];
+    const repoBranchPairs = privateRepositories.map((repository, index) => ({
+      repository,
+      branch: branches[index]
+    }));
+
+    for (const { repository, branch } of repoBranchPairs) {
+      const targetDir = `${tutorRoot}/env/build/openedx/requirements/${repository}`;
+      await gitClone(repository, branch, targetDir, options, ghAccessToken);
+      requirements.push(`-e ./${repository}`);
+    }
+  
+    if (requirements.length > 0) {
+        try {
+            fs.appendFileSync(privateTxtPath, `${requirements.join('\n')}\n`);
+        } catch (error) {
+            core.error(`Failed to write to ${privateTxtPath}`);
+            throw error;
+        }
+    }
```

Comment:
You already validated this when checking that privateRepositories was not empty and matched branches. @Serafin-dev 
```suggestion
    try {
        fs.appendFileSync(privateTxtPath, `${requirements.join('\n')}\n`);
    } catch (error) {
        core.error(`Failed to write to ${privateTxtPath}`);
        throw error;
    }
```